core/include/drivers/pinctrl.h

```diff
@@ -0,0 +1,126 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2022, Microchip
+ */
+
+#ifndef __DRIVERS_PINCTRL_H
+#define __DRIVERS_PINCTRL_H
+
+#include <kernel/dt_driver.h>
+#include <sys/queue.h>
+#include <tee_api_types.h>
+
+enum pin_config_mode {
+	PIN_MODE_BIAS_DISABLE,
+	PIN_MODE_BIAS_PULL_UP,
+	PIN_MODE_BIAS_PULL_DOWN,
+};
+
+struct pinconf {
+	const struct pinctrl_ops *ops;
```

Comment:
instead of passing `ops` in each pinconf, it could be made common in `struct pinctrl_state`

Reply:
ok